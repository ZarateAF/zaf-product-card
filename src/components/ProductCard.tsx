import React, { createContext, CSSProperties } from "react";
import useProduct from "../hooks/useProduct";
import styles from "../styles/styles.module.css";
import {
  Product,
  onChangeArgs,
  ProductContextProps,
  InitialValues,
  ProductCardHandlers,
} from "../interfaces/ProductInterfaces";

export interface Props {
  product: Product;
  children: (args: ProductCardHandlers) => JSX.Element;
  className?: string;
  style?: CSSProperties;
  onChange?: (args: onChangeArgs) => void;
  value?: number;
  initialValues?: InitialValues;
}

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

export const ProductCard = ({
  children,
  product,
  className,
  style,
  value,
  initialValues,
  onChange,
  
}: Props) => {
  const { counter, maxCount, isMaxCountReached, reset, increaseBy } = useProduct({
    onChange,
    product,
    value,
    initialValues,
  });

  return (
    <Provider
      value={{
        counter,
        maxCount,
        product,
        increaseBy,
      }}
    >
      <div className={`${styles.productCard} ${className}`} style={style}>
        {children({
          count: counter,
          isMaxCountReached,
          maxCount: initialValues?.maxCount,
          product,
          
          increaseBy,
          reset,
        })}
      </div>
    </Provider>
  );
};
