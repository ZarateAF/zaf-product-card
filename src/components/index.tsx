
import { ProductTitle } from "./ProductTitle";
import { ProductImage } from "./ProductImage";
import { ProductButtons } from "./ProductButtons";
import { ProductCard as ProductCardHOC } from "./ProductCard";
import { ProductCardHOCProps } from "../interfaces/ProductInterfaces";

export { ProductTitle } from "./ProductTitle";
export { ProductImage } from "./ProductImage";
export { ProductButtons } from "./ProductButtons";

export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
    Image: ProductImage,
    Title: ProductTitle,
    Buttons: ProductButtons,
})

export default ProductCard