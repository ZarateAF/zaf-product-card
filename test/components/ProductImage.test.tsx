import React from "react";
import renderer from "react-test-renderer";
import { ProductImage } from "../../src/components";
import { ProductCard } from "../../src/components/ProductCard";
import { product2 } from "../data/products";

describe('ProductImage', () => {
    test('should display the component successful', () => {
        const wrapper = renderer.create(
            <ProductImage img="./image/none.png" className="custom-class" />
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('should show a component with product image', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )
        expect(wrapper.toJSON()).toMatchSnapshot();        
    })
})