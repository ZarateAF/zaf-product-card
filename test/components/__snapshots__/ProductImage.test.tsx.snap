// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should display the component successful 1`] = `
<img
  alt="coffee"
  className="productImg custom-class"
  src="./image/none.png"
/>
`;

exports[`ProductImage should show a component with product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="coffee"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
